datafusion/common/src/dfschema.rs

```diff
@@ -119,74 +120,113 @@ impl DFSchema {
     /// Creates an empty `DFSchema`
     pub fn empty() -> Self {
         Self {
-            fields: vec![],
-            metadata: HashMap::new(),
+            inner: Arc::new(Schema::new([])),
+            field_qualifiers: vec![],
             functional_dependencies: FunctionalDependencies::empty(),
         }
     }
 
-    /// Create a new `DFSchema`
+    /// Create a new `DFSchema` from an Arrow schema
     pub fn new_with_metadata(
-        fields: Vec<DFField>,
+        fields: Vec<Field>,
```

Comment:
I think we can use`Fields` here, since Schema::new_with_metadata also expect `Feilds`.